src/features/gallery/GalleryProvider.tsx

```diff
@@ -86,8 +89,10 @@ export default function GalleryProvider({ children }: GalleryProviderProps) {
       animationType?: PreparedPhotoSwipeOptions["showHideAnimationType"]
     ) => {
       if (lightboxRef.current) return;
+      if (!post) return;
```

Comment:
Please don't guard here, some non post components may use this. Instead, `if(post) { do the dispatching }`